File: web/api-client/types/task.ts
/** 异步任务与通知类型。 */

/** 异步任务类型。 */
export type TaskType = "UPLOAD" | "INGEST" | "INDEX_BUILD" | "SYNC" | "DELETE" | "EXPORT";
export type TaskStatus = "PENDING" | "RUNNING" | "SUCCEEDED" | "FAILED" | "CANCELLING" | "CANCELLED";

export interface Task {
  id: number;
  type: TaskType;
  status: TaskStatus;
  title: string;
  /** 0-100,运行中/完成态展示进度。 */
  progress: number;
  /** 关联资源(文档/知识库/连接器),可选。 */
  resourceType?: string;
  resourceId?: string;
  startedAt: string;
  finishedAt: string | null;
  message?: string;
}

/** 通知类型。 */
export type NotificationKind = "TASK_DONE" | "TASK_FAILED" | "REVIEW_TODO" | "QUOTA_WARN" | "SYSTEM";
export type NotificationLevel = "info" | "success" | "warning" | "error";

export interface NotificationItem {
  id: number;
  kind: NotificationKind;
  level: NotificationLevel;
  title: string;
  body: string;
  read: boolean;
  createdAt: string;
  /** 可选跳转路径。 */
  href?: string;
}

File: service/src/main/java/com/ragkb/service/common/Task.java
package com.ragkb.service.common;

import java.time.Instant;

/**
 * 通用异步任务（对齐前端 Task 类型）。
 *
 * <p>用于克隆 / 删除 / 索引构建 / 同步 / 解析等异步操作：控制器返回 202，
 * 前端可轮询 {@code GET /api/v1/tasks/{taskId}} 获取进度。
 */
public record Task(
        String id,
        String type,
        String status,
        String title,
        int progress,
        String resourceType,
        String resourceId,
        Instant startedAt,
        Instant finishedAt,
        String message) {

    public static Task of(String id, String type, String status, String title, int progress) {
        return new Task(id, type, status, title, progress, null, null, Instant.now(), null, null);
    }
}
